source/grid/manifold.cc

```diff
@@ -87,6 +87,68 @@ get_new_point (const Quadrature<spacedim> &quad) const
   return p;
 }
 
+template <int dim, int spacedim>
+Point<spacedim>
+Manifold<dim, spacedim>::
+get_new_point (const std::vector<Point<spacedim> > &surrounding_points,
+               const std::vector<double>           &weights) const
+{
```

Comment:
Can't you redirect the now-deprecated `get_new_point` with quadrature argument to this function by calling `get_new_point(quad.get_points(), quad.get_weights())`? The same on the `get_new_point` methods of the derived classes below.